### PR TITLE
HPCC-16936 Add file size to XRef Lost Files Query Result

### PR DESCRIPTION
**Pull request by wangkx**
Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

### CODE REVIEW

**Comment by wangkx** [comment]
@jakesmith please review.

File: esp/services/ws_dfu/ws_dfuXRefService.cpp [hunk 1]
```diff
@@ -195,16 +196,52 @@ IXRefFilesNode* CWsDfuXRefEx::getFileNodeInterface(IXRefNode& XRefNode,const cha
...
+
+    Owned<IPropertyTreeIterator> iter = lostFilesQueryResult->getElements("File");
+    ForEach(*iter)
+    {
+        IPropertyTree& item = iter->query();
+        const char* fileName = item.queryProp("Name");
+        if (!fileName || !*fileName)
+            continue;
+
+        Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(fileName, userdesc, false, false, true); // lock super-owners
```
Comment (jakesmith) [inline esp/services/ws_dfu/ws_dfuXRefService.cpp]: Am not sure why the lockSuperOwner = true is here?

Also, I suspect it should have a timeout of 0 (fail if locked) and capture the failure and continue, providing a warning that can't recover lost file as locked.
NB: it shouldn't be in use/locked, but it's better than stalling/failing here.

**Comment by jakesmith** [comment]
@wangkx - one minor comment

**Comment by wangkx** [comment]
@jakesmith please review my changes.

File: esp/services/ws_dfu/ws_dfuXRefService.cpp [hunk 2]
```diff
@@ -223,9 +223,17 @@ void CWsDfuXRefEx::readLostFileQueryResult(IEspContext &context, StringBuffer& b
...
-            item.addPropInt64("Size", queryDistributedFileSystem().getSize(df));
+        try
+        {
+            Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(fileName, userdesc, false, false, false, NULL, 0);
+            if(df)
+                item.addPropInt64("Size", queryDistributedFileSystem().getSize(df));
+        }
+        catch(IException* e)
+        {
+            item.addProp("Status", "Warning: this file may be locked now. It can't be recovered as locked.");
```
Comment (jakesmith) [inline esp/services/ws_dfu/ws_dfuXRefService.cpp]: Probably worth logging the error as well, in case it's something other than a lock timeout..

**Comment by jakesmith** [comment]
@wangkx - could be merged now, but 1 more minor comment.

**Comment by wangkx** [comment]
@jakesmith I just added the code to log the exception. Please review.

File: esp/services/ws_dfu/ws_dfuXRefService.cpp [hunk 3]
```diff
@@ -195,16 +196,62 @@ IXRefFilesNode* CWsDfuXRefEx::getFileNodeInterface(IXRefNode& XRefNode,const cha
...
+            StringBuffer eMsg;
+            PROGLOG("Exception in readLostFileQueryResult(): %s", e->errorMessage(eMsg).str());
```
Comment (jakesmith) [inline esp/services/ws_dfu/ws_dfuXRefService.cpp]: trivial: EXCLOG is good for logging exceptions.

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge.

**Pull request merged**